### ISSUE TITLE
Pop over not closing on backdrop click when open prop is passed

### ISSUE THREAD

**Issue by mohitPanwar1996** [shadcn-ui/ui#2798]
Hey i am using a popover and it is not closing itself on the backdrop click when the open prop is passed to the popover is there any way to close the popover on the backdrop click.

**Comment by johurul-haque** [comment]
I am assuming you are trying to manually control the state of the popover. Here is how I approach this
```tsx
const [isOpen, setIsOpen] = useState(true)

<Popover open={isOpen} onOpenChange={setIsOpen}>
// ...
</Popover>
```

Having `useState(true)` means the popover will be open initially. Then `onOpenChange` will pass a boolean value to the setter function every time the state of the popover changes.

**Comment by shadcn** [comment]
This issue has been automatically closed because it received no activity for a while. If you think it was closed by accident, please leave a comment. Thank you.

**Comment by josheverett** [comment]
@johurul-haque I'm trying your solution, but this doesn't seem to work if the `useState` is `false` at first.

When you want to open the popover and call `setIsOpen(true)` the `onOpenChange` callback is (oddly) passed `false`, and thus immediately sets the state back to `false`. The result is that the popover shows for a split second before hiding itself.

**Comment by josheverett** [comment]
Update: I found that using `setTimeout` for the initial `setIsOpen(true)` call with a 200ms timeout hacks around the issue.

I also found that *at no point* is `onOpenChange` called with `true`. 

I have a popover the needs to be anchored to a specific element, but is opened programmatically in a number of places. To get that working, I used this method with `PopoverAnchor`:

https://github.com/shadcn-ui/ui/discussions/1684

So seems that `onOpenChange` *never* being called with `true` is a bug here.
